tri_mocap_common/CMakeLists.txt.ros2

```diff
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 3.5)
+project(tri_mocap_common)
+
+find_package(ament_cmake REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+
+find_package(geometry_msgs REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(tf2_msgs REQUIRED)
+
+## We don't depend on Drake, but we do use different build flags if present.
+find_package(drake QUIET)
+
+## Generate messages in the 'msg' folder
+rosidl_generate_interfaces(${PROJECT_NAME}_msgs
+  msg/MocapMarker.msg
+  msg/MocapMarkerArray.msg
+  msg/MocapObject.msg
+  msg/MocapSegment.msg
+  msg/MocapState.msg
+  DEPENDENCIES geometry_msgs std_msgs
+)
+
+###########
+## Build ##
+###########
+
+## Build options
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+cmake_policy(SET CMP0069 NEW)
+
+add_compile_options(-std=c++14)
+add_compile_options(-Wall)
+add_compile_options(-Wextra)
+add_compile_options(-Werror)
+add_compile_options(-Wconversion)
+add_compile_options(-Wshadow)
+add_compile_options(-O3)
+add_compile_options(-g)
+add_compile_options(-Werror=non-virtual-dtor)
+add_compile_options(-Wold-style-cast)
+add_compile_options(-Wpessimizing-move)
+add_compile_options(-Wuninitialized)
+
+if(drake_FOUND)
+    message(STATUS "Drake found, disabling -march=native")
+else()
+    message(STATUS "Drake NOT found, enabling -march=native")
+    add_compile_options(-march=native)
+endif()
+
+## Declare ROS C++ components
+add_library(
+  ${PROJECT_NAME}_components SHARED src/mocap_tf_publisher_node.ros2.cpp)
+ament_target_dependencies(
+  ${PROJECT_NAME}_components rclcpp rclcpp_components tf2_msgs)
```

Comment:
I'd recommend `target_link_libraries()` over `ament_target_dependencies()` now that ROS 2 is more consistent about using modern CMake targets. In this case it allows this library to avoid unnecessarily depending on the component manager.

```cmake
target_link_libraries(
    ${PROJECT_NAME}_components PUBLIC
    rclcpp::rclcpp
    rclcpp_components::component
    ${tf2_msgs_TARGETS})
```

Reply:
How recent is this change? @hidmic followed this structure with the other ports in this repo, and we'd like for the drivers to mostly work on Galactic.

Reply:
Ah, Galactic doesn't have the `rclcpp_components::component` target, so `ament_target_dependencies()` is still fine.